src/include/migraphx/op/scatter.hpp

```diff
@@ -33,29 +44,38 @@ struct scatter
         return {{"normalize_axes", normalize}};
     }
 
-    std::string name() const { return "scatter"; }
-
     shape normalize_compute_shape(std::vector<shape> inputs) const
     {
         check_shapes{inputs, *this}.has(3).standard();
-        return inputs.front();
+        // If non-packed, this converts to a packed output while preserving permutation of tensor
+        return inputs.front().with_lens(inputs.front().lens());
```

Comment:
This is conflict with the standard input shape requirement. We can just return the inputs.front() here.

Reply:
Actually, `.standard()` should be removed. The ref implementations should not require standard shape whenever possible.

Reply:
Can you clarify how the input shape requirement is set (i.e. how would I have known?)  Does it have to do with scatter being listed in parse_generic_op::needs_contiguous()?

Reply:
Considering the plan that we will refactor these operators, it is fine to put a standard shape here. Otherwise, we need to revisit the implementation to ensure they are correct.

Reply:
Sorry, this is the line for the standard input shape requirement 

```
check_shapes{inputs, *this}.has(3).standard();
```

This means it needs to have 3 inputs, and each one of them is with standard input shape.  You can check the [file](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/check_shapes.hpp)
for different shape requirements.